clients/privacy-center/features/consent/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { ConfigConsentOption } from "~/config/types";
+
+import {
+  ConsentItem,
+  ApiUserConsents,
+  ApiUserConsent,
+  CookieKeyConsent,
+} from "./types";
+
+export const makeConsentItems = (
+  data: ApiUserConsents,
+  consentOptions: ConfigConsentOption[]
+): ConsentItem[] => {
+  if (data.consent) {
+    const newConsentItems: ConsentItem[] = [];
+    const userConsentMap: { [key: string]: ApiUserConsent } = {};
+    data.consent.forEach((option) => {
+      const key = option.data_use;
+      userConsentMap[key] = option;
+    });
+    consentOptions.forEach((d) => {
+      if (d.fidesDataUseKey in userConsentMap) {
+        const currentConsent = userConsentMap[d.fidesDataUseKey];
+
+        newConsentItems.push({
+          consentValue: currentConsent.opt_in,
+          defaultValue: d.default ? d.default : false,
+          description: currentConsent.data_use_description
+            ? currentConsent.data_use_description
+            : d.description,
+          fidesDataUseKey: currentConsent.data_use,
+          highlight: d.highlight ?? false,
+          name: d.name,
+          url: d.url,
+          cookieKeys: d.cookieKeys ?? [],
+        });
+      } else {
+        newConsentItems.push({
+          fidesDataUseKey: d.fidesDataUseKey,
+          name: d.name,
+          description: d.description,
+          highlight: d.highlight ?? false,
+          url: d.url,
+          defaultValue: d.default ? d.default : false,
+          cookieKeys: d.cookieKeys ?? [],
+        });
+      }
+    });
+
+    return newConsentItems;
+  }
+
+  return consentOptions.map((option) => ({
+    fidesDataUseKey: option.fidesDataUseKey,
+    name: option.name,
+    description: option.description,
+    highlight: option.highlight ?? false,
+    url: option.url,
+    defaultValue: option.default ? option.default : false,
+    cookieKeys: option.cookieKeys ?? [],
+  }));
+};
+
+export const makeCookieKeyConsent = (
+  consentItems: ConsentItem[]
+): CookieKeyConsent => {
+  const cookieKeyConsent: CookieKeyConsent = {};
+  consentItems.forEach((item) => {
+    const consent =
+      item.consentValue === undefined ? item.defaultValue : item.consentValue;
```

Comment:
Should it read `item.consentValue ? item.consentValue : item.defaultValue`;

Reply:
These are booleans, so the explicit undefined check is need to distinguish between a missing value (no user selection) and a value set to false. The default only applies if the user didn't make a choice.